### ISSUE TITLE
`no-unused-expressions` false positive

### ISSUE THREAD

**Issue by feross** [eslint/eslint#8268]
**Tell us about your environment**

* **ESLint Version:** 3.16.1
* **Node Version:** 6.9.5
* **npm Version:** 3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

```
"no-unused-expressions": ["error", {
  "allowShortCircuit": true,
  "allowTernary": true
}]
```

**Intro**

Template strings are a way to interpolate variables with strings. It's nice, and often more readable than traditional string concatenation

Tagged template strings take this behavior a bit further, and allow functions to be called as if they were strings - but instead the functions receive multiple strings, and a trailing array of positional arguments that the have been used for interpolation. This allows operating on the string and adding all sorts of interesting behavior.

**What did you do? Please include the actual source code causing the issue.**

```js
var css = require('sheetify')

// this is a static expression
`
  .js-yellow { color: #f7df1e }
  .bg-js-yellow { background-color: #f7df1e }
`

// this is a function call, which in this case has side effects
css`
  .js-yellow { color: #f7df1e }
  .bg-js-yellow { background-color: #f7df1e }
`
```

**What did you expect to happen?**

The first case should be an error, because the string is unused. But the second case (a tagged template string) should not cause an error.

**What actually happened? Please include the actual, raw output from ESLint.**

```
index.js:14:1: Expected an assignment or function call and instead saw an expression.
```

The second case was considered a static expression, so it caused an error. This is not true, since it's effectively a function call and it has side effects in this case.

**Comment by not-an-aardvark** [comment]
I think this is a duplicate of https://github.com/eslint/eslint/issues/7632. It's fixed by https://github.com/eslint/eslint/pull/8253.

**Comment by platinumazure** [comment]
Hi @feross, thanks for the issue!

This will be resolved by #8253 once it is merged. Hopefully it will make the next release tomorrow. Thanks!

**Comment by feross** [comment]
Excellent. Sorry for the noise! Closing now.
